src/js/processor.js

```diff
@@ -0,0 +1,285 @@
+        return;
+      }
+
+      if (event.code == 'Enter') {
+        tArea.innerHTML += '\n';
+        return;
+      }
+
+      if (event.metaKey) {
+        event.preventDefault();
+        return;
+      }
+
+      // if arrow key is pressed
+      if (event.code.match(constants.arrowRegExp)) {
+        if (event.code == 'ArrowUp') {
+          tArea.innerHTML += '↑';
+          return;
+        }
+
+        if (event.code == 'ArrowDown') {
+          tArea.innerHTML += '↓';
+          return;
+        }
+
+        if (event.code == 'ArrowLeft') {
+          tArea.innerHTML += '←';
+          return;
+        }
+
+        if (event.code == 'ArrowRight') {
+          tArea.innerHTML += '→';
+          return;
+        }
+        return;
+      }
+
+      // if any other key is pressed
+      tArea.innerHTML += event.key;
+    })
+}
+
+export const virtualClickHandler = function () {
+  let kb = document.querySelector('.keyboard-block');
+  let tArea = document.querySelector('.textarea-block');
+
+  kb.onclick = function (event) {
+    let target = event.target;
+
+    if (!target.classList.contains('keyboard-button')) return;
+
+    if (target.innerHTML == 'Shift') {
+      return;
+    }
+
+    if (target.innerHTML == 'Ctrl') {
+      return;
+    }
+
+    if (target.innerHTML == 'Alt') {
+      return;
+    }
+
+    if (target.innerHTML == 'Tab') {
+      tArea.innerHTML += '    ';
+      return;
+    }
+
+    if (target.innerHTML == '') {
+      tArea.innerHTML += ' ';
+    }
+
+    if (target.innerHTML == 'Caps Lock') {
+      changeRegister();
+      return;
+    }
+
+    if (target.innerHTML == 'Backspace') {
+      let tAreaText = document.querySelector('.textarea-block').innerHTML;
+      tAreaText = tAreaText.substring(0, tAreaText.length - 1);
+      tArea.innerHTML = tAreaText;
+      return;
+    }
+
+    if (target.innerHTML == 'Del') {
+      let tAreaText = document.querySelector('.textarea-block').innerHTML;
+      tAreaText = tAreaText.substring(1, tAreaText.length);
+      tArea.innerHTML = tAreaText;
+      return;
+    }
+
+    if (target.innerHTML == 'Enter') {
+      tArea.innerHTML += '\n';
+      return;
+    }
+
+    // if arrow key is pressed
+    if (target.innerHTML.match(constants.arrowRegExp)) {
+      if (target.innerHTML == 'ArrowUp') {
+        tArea.innerHTML += '↑';
+        return;
+      }
+
+      if (target.innerHTML == 'ArrowDown') {
+        tArea.innerHTML += '↓';
+        return;
+      }
+
+      if (target.innerHTML == 'ArrowLeft') {
+        tArea.innerHTML += '←';
+        return;
+      }
+
+      if (target.innerHTML == 'ArrowRight') {
+        tArea.innerHTML += '→';
+        return;
+      }
+      return;
+    }
+
+    if (target.innerHTML == 'Info') {
+      let infoBlockButton = document.querySelector('.info-block-button');
+      info.show();
+
+      infoBlockButton.addEventListener('click', () => {
+        info.hide();
+      });
+      return;
+    }
+
```

Comment:
maybe it is better remember a link to info block once and reuse it instead of searching it into DOM tree

Reply:
Done